web/packages/teleterm/src/ui/DocumentCluster/UnifiedResources.tsx

```diff
@@ -133,14 +155,34 @@ export function UnifiedResources(props: UnifiedResourcesProps) {
     return cleanup;
   }, [onResourcesRefreshRequest, fetch, clear]);
 
-  function onParamsChange(newParams: UnifiedResourcesQueryParams): void {
+  const [prevParams, setPrevParams] = useState(props.queryParams);
+  if (prevParams !== props.queryParams) {
+    setPrevParams(props.queryParams);
     clear();
-    setParams(newParams);
+  }
+
+  function onParamsChange(newParams: UnifiedResourcesQueryParams): void {
```

Comment:
>I suspect that you had to add this because setParams used to live within this component but now query params are passed from documentService's state, hence why we cannot call clear from within onParamsChange anymore since by this point DocumentCluster is yet to be re-rendered. Is that true?

Correct.

I added `key` to the child component, it seems to work well. 
I used `JSON.stringify` to get the key string, I think it is fine. An alternative would be to craft the string manually, but I don't think we have to care about properties reordering?

Reply:
Yeah, I think `JSON.stringify` should work!